tap_kit/client.py

```diff
@@ -31,18 +33,26 @@ def requests_method(method, request_config, body):
 
     @backoff.on_exception(backoff.expo,
                           RateLimitException,
+                          Exception,
                           max_tries=10,
                           factor=2)
     def make_request(self, request_config, body=None, method='GET'):
         LOGGER.info("Making {} request to {}".format(
             method, request_config['url']))
 
         with singer.metrics.Timer('request_duration', {}) as timer:
-            response = self.requests_method(method, request_config, body)
+            try:
+                response = self.requests_method(method, request_config, body)
+            except Exception:
```

Comment:
You can remove this try/except block, it's unnecessary because `response.raise_for_status()` will handle this for you.

Reply:
So I thought I would need this part for errors like this

requests.exceptions.SSLError: HTTPSConnectionPool(host='1stdibs.my.salesforce.com', port=443): Max retries exceeded with url: /services/data/v37.0/sobjects/s4g_currencies__LogCurrency__c/describe (Caused by SSLError(SSLError("bad handshake: SysCallError(-1, 'Unexpected EOF')")))

or 

requests.exceptions.ConnectionError: HTTPSConnectionPool(host='pipes.automation.simondata.net', port=443): Max retries exceeded with url: /api/python?tree=jobs%5Bname%2Ccolor%2Curl%5D (Caused by ReadTimeoutError("HTTPSConnectionPool(host='pipes.automation.simondata.net', port=443): Read timed out. (read timeout=10)",))

Reply:
I was thinking creating a try catch for these errors and then having another if statement below to catch the status code errors we want to look for

Reply:
Because you're catching all Exceptions (`except Exception`) and then raising an Exception on the next line, code execution will stop here. You won't even make it to your status code checks below. `response.raise_for_status()` will `raise` any errors that are not explicitly handled in your status code checks.

Reply:
SSLError won't return a status code, it's at the transport layer rather than the protocol layer, there's a class of errors that you can retry in the decorator, see https://github.com/Radico/web/blob/develop/simon/integrations/mailchimp.py#L53

Reply:
Thanks Tom! This was exactly what I was looking for!